### PR TITLE
Integrating GPQA Dataset

### PR DESCRIPTION
**Pull request by vshrivas**

### CODE REVIEW

**Comment by vshrivas** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: eureka_ml_insights/data_utils/transform.py [hunk 1]
```diff
@@ -180,6 +181,49 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
...
+        def shuffle_row(row):
+            row[self.columns] = np.random.permutation(row[self.columns].values)
+            return row
+
+        df = df.apply(shuffle_row, axis=1)
+        return df
+
+
+@dataclass
+class ColumnMatchMap(MultiColumnTransform):
```
[the remark this answers is not in the document]
Comment (vshrivas) [reply eureka_ml_insights/data_utils/transform.py:205]: I think this depends on the definition of a MultiColumnTransform. Multiple columns are not getting updated, but we do pass in a list of multiple columns. Specifically the value in key_col is compared to the values in each of the columns passed in, and the value in new_col is assigned accordingly. For example, in the GPQA pipeline this is how we call the transform:
```
ColumnMatchMap(
                                new_col="ground_truth", key_col="Correct Answer", columns=["A", "B", "C", "D"]
                            ),
```
If this doesn't fit the definition of a MultiColumnTransform, I can change the class to inherit from DFTransformBase instead.

File: eureka_ml_insights/data_utils/transform.py [hunk 2]
```diff
@@ -180,6 +181,49 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
...
+    """Creates a new column indicating the name of the column that matches the value in the key column for each row.
+    E.g. for a row, if value of key_col matches value of 'A' column, new_col will contain the value 'A'."""
+
+    key_col: str
+    new_col: str
+    columns: List[str] | str
```
[the remark this answers is not in the document]
Comment (vshrivas) [reply eureka_ml_insights/data_utils/transform.py:211]: Thanks, will update to List[str]. Let me know what you think about wrt inheriting from MultiColumnTransform based on the other comment.

File: tests/pipeline_tests.py [hunk 3]
```diff
@@ -268,11 +269,23 @@ def configure_pipeline(self, resume_from=None):
         self.data_processing_comp.data_reader_config.init_args["split"] = "dev"
         self.data_processing_comp.data_reader_config.init_args["tasks"] = ["Math"]
 
-        self.inference_comp.data_loader_config.class_name = TestMMDataLoader
+        self.inference_comp.data_loader_config.class_name = TestDataLoader
         self.inference_comp.data_loader_config.init_args["n_iter"] = N_ITER
         return config
 
 
+class TEST_GPQA_PIPELINE(GPQA_Experiment_Pipeline):
```
[the remark this answers is not in the document]
Comment (vshrivas) [reply tests/pipeline_tests.py:277]: It seems like the decorator needs to be added to the function that runs this test -- GPQA_PipelineTest. That's where I added it.

File: eureka_ml_insights/configs/model_configs.py [hunk 4]
```diff
@@ -1,9 +1,9 @@
 """ This module contains config objects for the models used in the experiments. To use these configs, make sure to
```
[the remark this answers is not in the document]
Comment (vshrivas) [reply eureka_ml_insights/configs/model_configs.py:1]: Done

File: eureka_ml_insights/data_utils/transform.py [hunk 5]
```diff
@@ -180,6 +181,49 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
         return df
 
 
+@dataclass
+class ShuffleColumns(MultiColumnTransform):
+    """For a set of columns, shuffles the values across each row of these columns."""
```
[the remark this answers is not in the document]
Comment (vshrivas) [reply eureka_ml_insights/data_utils/transform.py:186]: Done!

File: tests/data_utils_tests/shuffle_columns_tests.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+from eureka_ml_insights.data_utils.transform import ShuffleColumns
+
+
+class TestShuffleColumns(unittest.TestCase):
+    def setUp(self):
+        # Seed the random number generator for reproducibility
+        np.random.seed(42)
+
+        self.values = [
```
[the remark this answers is not in the document]
Comment (vshrivas) [reply tests/data_utils_tests/shuffle_columns_tests.py]: Went with the second suggestion!

File: eureka_ml_insights/data_utils/transform.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (safooray) [reply eureka_ml_insights/data_utils/transform.py:211]: Definitely not a great fit as apparent by the amount of logic that is reused from MultiColumnTransform vs. overridden. You are reusing the validate method of MultiColumnTransform to validate if `columns` exist in the df, but you're not validating the same for key_col. Also you are having to fully override the transform method from MultiColumnTransform. So overall I think this inheritance is confusing for the reader.

**Review by safooray — APPROVED** [review]

**Review by safooray — APPROVED** [review]

**Pull request merged**